### PR TITLE
Visit confirmation security issue

### PR DESCRIPTION
**Pull request by vlasiak**
Resolves [Security hole: Visit confirmation should rely on its `token`, not `id`](https://github.com/pivorakmeetup/pivorak-web-app/issues/545)

### CODE REVIEW

File: README.md
Comment (floor114) [inline README.md:25]: Why do you remove it?

File: app/controllers/visit_requests_controller.rb
Comment (floor114) [inline app/controllers/visit_requests_controller.rb:34]: You don't need to do it here. Move it to `VisitRequest::FinalConfirmation` service. 
```ruby
VisitRequest::FinalConfirmation.call(params)
```

File: app/services/visit_request/final_confirmation.rb
Comment (floor114) [inline app/services/visit_request/final_confirmation.rb:5]: Think all should be as it was. Just add `visit_request` method here.
```ruby
def visit_request
   @visit_request ||= VisitRequest.find_by(token: params[:token])
end
```

**Review by floor114 — CHANGES_REQUESTED** [review]

File: README.md
Comment (vlasiak) [reply README.md:25]: Because there is no such file `config/environments/development.rb.example` in the repository.

File: app/controllers/visit_requests_controller.rb
Comment (vlasiak) [reply app/controllers/visit_requests_controller.rb:34]: Agree, that controller isn't the best place for this code, even though it is shared among the actions. Anyway, it was already placed here and it was the fastest way to achieve the goal.

File: app/services/visit_request/final_confirmation.rb
Comment (vlasiak) [reply app/services/visit_request/final_confirmation.rb:5]: Currently I see validations in this file useless. In case we leave them, service has to return some validation errors (not `nil`) and controller has to deal with them and present appropriate flash message. Thus, it requires more complex refactoring, imho.

File: app/services/visit_request/final_confirmation.rb
Comment (denys-medynskyi) [inline app/services/visit_request/final_confirmation.rb:16]: Maybe add `else` branch and fail with an error?

**Comment by denys-medynskyi** [comment]
Overall - 👍 🥇 job. Thanks for taking the initiative.
1. Please rebase to solve the conflicts.
2. Specs are failing - check them please.
3. If you need any assistance - ping me in Slack.

File: app/controllers/visit_requests_controller.rb
Comment (floor114) [reply app/controllers/visit_requests_controller.rb:34]: Sorry, my bad (misunderstood). Same for next comment.

File: app/services/visit_request/final_confirmation.rb
Comment (vlasiak) [reply app/services/visit_request/final_confirmation.rb:16]: Makes sense.

**Comment by vlasiak** [comment]
@floor114, @denysmedynskyi, check this out.

**Review by floor114 — DISMISSED** [review]

**Comment by floor114** [comment]
@vlasiak Could you squash commits before merging?

**Comment by vlasiak** [comment]
@floor114, done. But squashing has dismissed your previous review approval.

**Review by floor114 — APPROVED** [review]

**Pull request merged**
